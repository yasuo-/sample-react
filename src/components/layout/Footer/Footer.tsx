import React from 'react';

import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

import { PATH } from '@/routes/path';

/**
 * Footer
 */
export const Footer: React.FC = () => {
  const { t } = useTranslation('common');
  const copyRightName = '....inc';
  const copayRightYear = new Date().getFullYear();

  const linkStyle = 'text-xs font-light tracking-tighter text-white/35 hover:text-white/100';
  return (
    <footer className="mx-auto mt-60 w-full bg-mine-shaft-800">
      <div className="flex justify-center justify-items-center py-2.5">
        <div>
          <Link to={PATH.TERMS} className={linkStyle}>
            {t('footer.terms')}
          </Link>
        </div>
        <div className="mx-2.5">
          <Link to={PATH.PRIVACY} className={linkStyle}>
            {t('footer.privacy')}
          </Link>
        </div>
        <div>
          <Link to={PATH.CONTACT} className={linkStyle}>
            {t('footer.contact')}
          </Link>
        </div>
      </div>
      <div className="flex justify-center justify-items-center pb-2.5">
        <span className="text-xs font-light tracking-tighter text-white/35">
          &copy;{copyRightName} {copayRightYear}
        </span>
      </div>
    </footer>
  );
};

Footer.displayName = 'Footer';
