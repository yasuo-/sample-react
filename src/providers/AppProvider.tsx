import React from 'react';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ErrorBoundary } from 'react-error-boundary';
import { HelmetProvider, HelmetData } from 'react-helmet-async';
import { I18nextProvider } from 'react-i18next';

import { ErrorFallback } from '@/components/error';
import i18n from '@/lib/locale/i18n';

type AppProviderProps = {
  children: React.ReactNode;
};

// Create a client
const queryClient = new QueryClient();
const helmetContext = new HelmetData({});

/**
 * AppProvider
 * @param children
 * @returns {JSX.Element}
 */
export const AppProvider: React.FC<AppProviderProps> = ({ children }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <HelmetProvider context={helmetContext}>
          <ErrorBoundary FallbackComponent={ErrorFallback}>{children}</ErrorBoundary>
        </HelmetProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};
