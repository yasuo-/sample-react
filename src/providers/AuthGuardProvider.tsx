import React, { createContext, useContext } from 'react';

export type AuthGuardContextType = {
  accessToken: string | null;
  signIn: (accessToken: string) => void;
  signOut: () => void;
};

type AuthGuardProviderProps = {
  children: React.ReactNode;
};

/**
 * AuthGuardContext
 */
const AuthGuardContext = createContext<AuthGuardContextType>({} as AuthGuardContextType);

/**
 * useAuthGuardContext
 */
export const useAuthGuardContext = (): AuthGuardContextType =>
  useContext<AuthGuardContextType>(AuthGuardContext);

/**
 * AuthGuardProvider
 * @param AuthGuardProviderProps {children: React.ReactNode}
 * @constructor
 */
export const AuthGuardProvider: React.FC<AuthGuardProviderProps> = ({ children }) => {
  const [accessToken, setAccessToken] = React.useState<string | null>(null);
  /**
   * signIn
   * @param accessToken {string} access token
   */
  const signIn = (accessToken: string) => {
    setAccessToken(accessToken);
  };
  /**
   * signOut
   */
  const signOut = () => {
    setAccessToken(null);
  };

  return (
    <AuthGuardContext.Provider value={{ accessToken, signIn, signOut }}>
      {children}
    </AuthGuardContext.Provider>
  );
};
