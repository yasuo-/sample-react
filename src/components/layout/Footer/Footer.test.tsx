import React from 'react';

import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Footer } from './Footer';

import { AppProvider } from '@/providers';

/**
 * renderWithRouter
 * helper function to render component with router
 * @param component
 */
const renderWithRouter = (component: React.ReactNode) => {
  return {
    ...render(
      <AppProvider>
        <MemoryRouter>{component}</MemoryRouter>
      </AppProvider>
    ),
  };
};

describe('Footer Components', () => {
  it('displayName is Footer', () => {
    expect(Footer.displayName).toBe('Footer');
  });

  it('should take a snapshot', () => {
    const { asFragment } = renderWithRouter(<Footer />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders Link terms', () => {
    renderWithRouter(<Footer />);
    const screenElement = screen.getByText(/利用規約/i);
    expect(screenElement).toBeInTheDocument();

    // const linkElement = screen.getByTestId('terms');
    // userEvent.click(linkElement);
    // expect(window.location.pathname).toEqual(PATH.TERMS);
  });

  it('renders text privacy', () => {
    renderWithRouter(<Footer />);
    const screenElement = screen.getByText(/プライバシーポリシー/i);
    expect(screenElement).toBeInTheDocument();

    //const linkElement = screen.getByTestId('privacy');
    // userEvent.click(linkElement);
    // expect(window.location.pathname).toEqual(PATH.PRIVACY);
  });

  it('renders text contact', () => {
    renderWithRouter(<Footer />);
    const screenElement = screen.getByText(/お問い合わせ/i);
    expect(screenElement).toBeInTheDocument();

    // const linkElement = screen.getByTestId('contact');
    // userEvent.click(linkElement);
    // expect(window.location.pathname).toEqual(PATH.CONTACT);
  });
});
