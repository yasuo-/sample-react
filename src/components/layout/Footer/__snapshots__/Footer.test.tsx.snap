// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Components should take a snapshot 1`] = `[Function]`;
